### ISSUE TITLE
[s3] http: superfluous response.WriteHeader

### ISSUE THREAD

**Issue by kmlebedev** [chrislusf/seaweedfs#1838]
**Describe the bug**
[s3] http: superfluous response.WriteHeader after `raft.Server: Not current leader` per request 
```
I0225 09:43:38     1 masterclient.go:120] filer masterClient failed to receive from fast-master-0.s3-fast-master-direct.service.dcix.consul:9333: rpc error: code = Unavailable desc = transport is closing
2021/02/25 09:43:38 http: superfluous response.WriteHeader call from github.com/chrislusf/seaweedfs/weed/server.processRangeRequest (common.go:283)
E0225 09:43:38     1 filer_server_handlers_write.go:42] failing to assign a file id: failed to parse master : server should have hostname:port format:
I0225 09:43:38     1 common.go:53] response method:PUT URL:/buckets/registry/docker/registry/v2/repositories/b2c/credits-admin/_uploads/71e12cb4-5c6e-405e-a6ca-1d491ec79441/hashstates/sha256/15728640 with httpStatus:500 and JSON:{"error":"failed to parse master : server should have hostname:port format: "}
E0225 09:43:38     1 s3api_object_handlers.go:330] upload to filer error: failed to parse master : server should have hostname:port format:
2021/02/25 09:43:38 http: superfluous response.WriteHeader call from github.com/chrislusf/seaweedfs/weed/server.processRangeRequest (common.go:283)
2021/02/25 09:43:38 http: superfluous response.WriteHeader call from github.com/chrislusf/seaweedfs/weed/server.processRangeRequest (common.go:283)	
```

https://github.com/chrislusf/seaweedfs/blob/master/weed/server/common.go#L284

**Expected behavior**
automatic recovery after loss of master
If there is no content, the S3 API returns 200 and 206 codes, need return 500

**Comment by kmlebedev** [comment]
```
filer_1    | E0226 09:48:34     1 common.go:284] failed to writeFn: volume 2 not found
filer_1    | 2021/02/26 09:48:34 http: superfluous response.WriteHeader call from github.com/chrislusf/seaweedfs/weed/server.processRangeRequest (common.go:285)
```

**Comment by kmlebedev** [comment]
incomplete data at the moment of a change of leader without http code 500
```
> GET /test_1/test.zip?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=some_access_key1%2F20210226%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210226T093424Z&X-Amz-Expires=86400&X-Amz-SignedHeaders=host&X-Amz-Signature=20452938422980c7a460eb68089869f7e43e09d74e0d168df17cdd6e6cbe0435 HTTP/1.1
> Host: 127.0.0.1:8333
> Range: bytes=0-10000
> User-Agent: curl/7.72.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 206 Partial Content
< Accept-Ranges: bytes
< Access-Control-Expose-Headers: Content-Disposition
< Content-Disposition: inline; filename="test.zip"
< Content-Length: 10001
< Content-Range: bytes 0-10000/10340419
< Content-Type: application/zip
< Date: Fri, 26 Feb 2021 09:50:12 GMT
< Etag: "e2e9ebf624eba4d3b68f1fc19e889a68-2"
< Last-Modified: Fri, 26 Feb 2021 09:33:07 GMT
< Server: SeaweedFS Filer 30GB 2.28
<
{ [3671 bytes data]
```

**Comment by kmlebedev** [comment]
If stop heartbeat for  volume server 
```
volumeServer.leave -node volume3:8083
stopped heartbeat in volume server volume3:8083. After a few seconds to drain traffic, it will be safe to stop the volume server.
```

```
> GET /test_1/txmlconnector-0.06.zip?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=some_access_key1%2F20210226%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210226T093424Z&X-Amz-Expires=86400&X-Amz-SignedHeaders=host&X-Amz-Signature=20452938422980c7a460eb68089869f7e43e09d74e0d168df17cdd6e6cbe0435 HTTP/1.1
> Host: 127.0.0.1:8333
> Range: bytes=0-10000
> User-Agent: curl/7.72.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 206 Partial Content
< Accept-Ranges: bytes
< Access-Control-Expose-Headers: Content-Disposition
< Content-Disposition: inline; filename="txmlconnector-0.06.zip"
< Content-Length: 10001
< Content-Range: bytes 0-10000/10340419
< Date: Fri, 26 Feb 2021 10:52:25 GMT
< Etag: "e2e9ebf624eba4d3b68f1fc19e889a68-2"
< Last-Modified: Fri, 26 Feb 2021 09:33:07 GMT
< Server: SeaweedFS Filer 30GB 2.28
<
{ [0 bytes data]
* transfer closed with 10001 bytes remaining to read
* Closing connection 0
```

**Comment by kmlebedev** [comment]
similarly for GET request 
```
curl -o /dev/null -sv "http://127.0.0.1:8333/test_1/test.zip?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=some_access_key1%2F20210226%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210226T093424Z&X-Amz-Expires=86400&X-Amz-SignedHeaders=host&X-Amz-Signature=20452938422980c7a460eb68089869f7e43e09d74e0d168df17cdd6e6cbe0435"
*   Trying 127.0.0.1:8333...
* Connected to 127.0.0.1 (127.0.0.1) port 8333 (#0)
> GET /test_1/txmlconnector-0.06.zip?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=some_access_key1%2F20210226%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210226T093424Z&X-Amz-Expires=86400&X-Amz-SignedHeaders=host&X-Amz-Signature=20452938422980c7a460eb68089869f7e43e09d74e0d168df17cdd6e6cbe0435 HTTP/1.1
> Host: 127.0.0.1:8333
> User-Agent: curl/7.72.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Accept-Ranges: bytes
< Access-Control-Expose-Headers: Content-Disposition
< Content-Disposition: inline; filename="test.zip"
< Content-Length: 10340419
< Date: Fri, 26 Feb 2021 11:23:38 GMT
< Etag: "e2e9ebf624eba4d3b68f1fc19e889a68-2"
< Last-Modified: Fri, 26 Feb 2021 09:33:07 GMT
< Server: SeaweedFS Filer 30GB 2.28
<
{ [0 bytes data]
* transfer closed with 10340419 bytes remaining to read
* Closing connection 0
```


logs 
```
s3_1       | I0226 11:23:38     1 auth_credentials.go:172] v4 auth type
s3_1       | I0226 11:23:38     1 auth_credentials.go:193] user name: test actions: [Read:test_1 Read:test_2 Read:test_3 Read:test_4 Read:test_5 Write:test_1 Write:test_2 Write:test_3 Write:test_4 Write:test_5 List:test_1 List:test_2 List:test_3 List:test_4 List:test_5 Tagging:test_1 Tagging:test_2 Tagging:test_3 Tagging:test_4 Tagging:test_5]
s3_1       | I0226 11:23:38     1 s3api_object_handlers.go:261] s3 proxying GET to http://filer:8888/buckets/test_1/test.zip
filer_1    | I0226 11:23:38     1 filerstore_wrapper.go:129] FindEntry /buckets/test_1/test.zip

```
